field.go

```diff
@@ -431,6 +431,12 @@ func (d dictObject) MarshalLogObject(enc zapcore.ObjectEncoder) error {
 	return nil
 }
 
+// DictObject exposes the [zapcore.ObjectMarshaler] for an array of [Field] to
+// use with functions like [Object] and [Objects].
```

Comment:
I would reword this a bit. Object/Objects aren't the only uses of this.

```suggestion
// DictObject constructs a [zapcore.ObjectMarshaler] with the given list of fields.
// The resulting object may be used as input to [Object], [Objects],
// or anywhere else that an object is expected.
```

Reply:
I reworded it a bit based on this feedback